openapi.yaml

```diff
@@ -8341,6 +8341,9 @@ components:
         * `chk-concurrency-max` - maximum number of check runs of the same type executing in parallel. We consider check runs executing in parallel if their time intervals overlap ≥ 50%.
         * `chk-concurrency` - average number of check runs of the same type executing in parallel; we choose the check run type that corresponds to the maximum concurrency for each day. We consider check runs executing in parallel if their time intervals overlap ≥ 50%.
         * `chk-elapsed-time-per-concurrency` - cumulative time spent on each concurrency level across check runs. Only allowed for histograms.
+        * `chk-suite-occupancy` - average ratio between summed time of check runs in a check suite and the product of the number of check runs with the maximum check run time in that check suite. 1 means ideal resource utilization, down to 0 for absolute inefficiency. For example, there are one slow check run that executes in 10x seconds, and N fast check runs that execute in x seconds, in the same check suite. The metric value will be (10x + Nx) / (10*(N+1)*x) → 0.1 if N → +∞.
+        * `chk-suite-critical-occupancy` - `chk-suite-occupancy` calculated for critical check runs only. A critical check run is a check run of type that at least once dominated the check suite execution time on the time interval. This is the measure of check suite "trebble" when there are several slowest check runs that elapse time with some standard deviation.
```

Comment:
Ok, it took me a moment to understand this one. If a check suite of 10 check runs is saturated by 2 different during the time range selected, this ratio is computed using as if the check suite counted those 2 check runs, right?

But I guess for most of the queries made by the users, there is going to be only one critical check run for each check suite, so the metric can tend towards 1 for 2 different reasons:
* most of the check suites have only 1 critical check run
* the critical check runs for each check suite have similar run times

If we want that metric to be useful, its interpretation needs to be straightforward: 1 good, 0 bad. According, I'm thinking of one user which starts to optimize its CI, and decrease the run time of the critical check run (good), at such a point when it starts to not be all the time critical -> multiple critical check runs -> the metric goes down -> the user feels he or she's doing a bad job.

Reply:
Agree let's see in practice how it behaves.